crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/prosecution_concluded/service/ReactivatedCaseDetectionService.java

```diff
@@ -0,0 +1,77 @@
+package uk.gov.justice.laa.crime.crowncourt.prosecution_concluded.service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
+import uk.gov.justice.laa.crime.crowncourt.dto.maatcourtdata.RepOrderCCOutcomeDTO;
+import uk.gov.justice.laa.crime.crowncourt.entity.ReactivatedProsecutionCase;
+import uk.gov.justice.laa.crime.crowncourt.prosecution_concluded.model.ProsecutionConcluded;
+import uk.gov.justice.laa.crime.crowncourt.repository.ReactivatedProsecutionCaseRepository;
+import uk.gov.justice.laa.crime.crowncourt.service.MaatCourtDataService;
+
+import java.time.LocalDateTime;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class ReactivatedCaseDetectionService {
+
+    private static final String PENDING = "PENDING";
+    private static final String SUPERSEDED = "SUPERSEDED";
+    private final MaatCourtDataService maatCourtDataService;
+    private final ReactivatedProsecutionCaseRepository reactivatedProsecutionCaseRepository;
+
+    public void processCase(ProsecutionConcluded prosecutionConcluded) {
+        if (!prosecutionConcluded.isConcluded()) {
+            boolean isCaseRecordExist = reactivatedProsecutionCaseRepository.existsByMaatIdAndReportingStatus(prosecutionConcluded.getMaatId(), PENDING);
+            if (!isCaseRecordExist){
+                createReactivatedCaseRecord(prosecutionConcluded);
```

Comment:
Also, what about if we received a new pending prosecution concluded, then the case worker creates a crown court outcome (even though we have only received a pending prosecution concluded - I understand this is possible from @mtac50 ), and then we receive another pending prosecution concluded - this would technically be a reactivation however would be untracked here.